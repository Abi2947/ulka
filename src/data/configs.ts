import "colors"
import path from "path"
import fs from "fs"

let ulkaConfigs: { [key: string]: string | any } = {}

try {
  const configExists = fs.existsSync(path.join(process.cwd(), "ulka-config.js"))

  if (configExists)
    ulkaConfigs = require(path.join(process.cwd(), "ulka-config.js"))
} catch (e) {
  console.log(`>> ${e.message}`.red)
  process.exit(0)
}

interface Config {
  buildPath: string
  pagesPath: string
  templatesPath: string
  contents: any[]
  plugins: any[]
}

const defConfig = {
  buildPath: "build",
  pagesPath: "pages",
  templatesPath: "templates",
  contents: [],
  plugins: []
}

const config: Config = { ...defConfig, ...ulkaConfigs }

export default config
