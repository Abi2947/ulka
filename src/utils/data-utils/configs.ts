import "colors"
import path from "path"
import fs from "fs"

let ulkaConfigs: { [key: string]: string | any } = {}

try {
  const configExists = fs.existsSync(path.join(process.cwd(), "ulka-config.js"))
  if (configExists)
    ulkaConfigs = require(path.join(process.cwd(), "ulka-config.js"))
} catch (e) {
  console.log(`>> ${e.message}`.red)
  process.exit(0)
}

const defConfig = {
  buildPath: "build",
  pagesPath: "pages",
  templatesPath: "templates",
  contents: [],
  plugins: []
}

const config = { ...defConfig, ...ulkaConfigs }

export default config
