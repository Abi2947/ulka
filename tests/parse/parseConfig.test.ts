import configs from '../../src/parse/parseConfig'

describe('PraseConfigs', () => {
  test('returns default configs', () => {
    expect(configs).toMatchSnapshot()
  })
})
