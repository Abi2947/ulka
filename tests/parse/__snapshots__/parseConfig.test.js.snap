// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PraseConfigs returns default configs 1`] = `
Object {
  "buildPath": "build",
  "contents": Object {
    "generatePath": "blog",
    "path": "contents",
    "postParse": Array [],
    "preParse": Array [],
    "template": "blog.ulka",
  },
  "pagesPath": "pages",
  "plugins": Array [],
  "siteMetaData": Object {
    "desription": "UlkaJs - static site generator",
    "title": "Ulka.Js",
  },
  "templatesPath": "templates",
}
`;
